x/evm/state/balance.go

```diff
@@ -23,13 +23,20 @@ func (s *DBImpl) SubBalance(evmAddr common.Address, amt *big.Int, reason tracing
 		evmAddr, _ = s.k.GetFeeCollectorAddress(s.ctx)
 	}
 
+	ctx := s.ctx
+
+	// this avoids emitting cosmos events for ephemeral bookkeeping transfers like send_native
```

Comment:
we probably need to swap the original event manager back after the logic below is done; otherwise the events that exist in the replaced event manager would be lost

Reply:
it creates a new context at this moment, so anyone referring the old context (the caller) keeps their existing events.